kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt

```diff
@@ -208,13 +208,17 @@ class CanvasState(val canvas: HTMLCanvasElement) {
         get() = v(width.toDouble(), height.toDouble())
     val context = creatures.context
     var valid = false
-    var shapes = ArrayList<Shape>()
+    var shapes = mutableListOf<Shape>()
     var selection: Shape? = null
     var dragOff = Vector()
     val interval = 1000 / 30
 
     init {
-        jq(canvas).mousedown {
+        fun fix(f: Element.(MouseEvent) -> Unit) : Element.(MouseEvent) -> Unit {
```

Comment:
AFAIK there is no way to avoid hacks at the moment. cc @bashor 

The way extension lambdas are translated has been changed. They used to get the receiver parameter from `this`. Now there is a dedicated parameter `$receiver` for that. That was done in order to use extension and regular lambdas interchangeably (e.g. so that (A).(B) -> C is equivalent to (A, B) -> C).